Merge back 'chore_release-8.7.0' into 'edge'

ddcc4 opened the pull request:
Lots of fixes!

ddcc4 committed:
fix(protocol-designer): include stack traces for console events to Sentry (#19572)

# Overview

We previously started capturing `console.assert()` failures to Sentry.
However, by default, these events do NOT report the stack trace to
Sentry. Without the stack trace, Sentry has no way to tell that
`console.assert()` failures are coming from the same place in the code,
so we end up with dozens of separate `expected to find a matching
labware def` reports in Sentry.

This PR configures Sentry to send the stack trace with these
`console.assert()` messages. After this change, Sentry should be able to
group them together if they are coming from the same place in the code.
https://docs.sentry.io/platforms/javascript/configuration/options/#attachStacktrace

## Test Plan and Hands on Testing

Run CI tests. But we'd have to deploy to see the effect.

## Risk assessment

Low.

jerader committed:
feat(app): enable py generation in QT (#19605)

There are a bunch of atomic command bugs in JSON right now and that's
because we didn't thoroughly test the commands. Generating python will
fix all the bugs in the atomic commands since we will be generating the
liquid class commands in python and it has been tested better for PD.

ddcc4 committed:
Merge back 'chore_release-pd-8.5.4' into 'chore_release-8.7.0' (#19607)

ddcc4 committed:
feat(shared-data): more liquid class values v2 changes (#19615)

# Overview

More tweaks to the liquid class `ethanol_80/2.json` values, courtesy
@meh-di and @thassyopinto. These were taken from the
`EXEC-1671-96-lc-feature-8_5` branch, commit c85017.

## Test Plan and Hands on Testing

Run the CI tests.

We'll do more hands-on testing with this.

## Risk assessment

Low.

ncdiehl11 committed:
fix(app): Quick Transfer settings overview style (#19614)

Fixes the text style for overview settings in QuickTransfer (from
regular -> semibold)

ncdiehl11 committed:
fix(protocol-designer): always show aspirate tip position button (#19619)

This PR fixes a bug wherein selecting trash for destination labware
incorrectly hid tip position from aspirate advanced settings. In this
case, we should only hide tip position from dispense settings.

Closes RQA-4686

jerader committed:
fix(app): pass in LABWAREV2_DO_NOT_LIST into QT (#19617)

closes RQA-4685

ncdiehl11 committed:
fix(app): fix tip positions in Quick Transfer (#19613)

In Quick Transfer, fix the tip position ranges and captions for
aspirate/dispense and submerge/retract, respecting position reference

Closes RQA-4668
Closes RQA-4669

jerader committed:
fix(step-generation): do not emit touchTip when blowout in touchTipDi… (#19622)

…sabled source

closes RQA-4684

TamarZanzouri committed:
fix(app): text fix for store shuttle empty ER error modal (#19621)

jerader committed:
feat(app): update QT version to 2.0.0 to include py generation (#19610)

ddcc4 committed:
fix(protocol-designer): handle missing pipette in pipetteTiprackAssignments (#19624)

# Overview

`NormalizedPipetteById.tiprackDefURI` is supposed to be an array of
tipracks. But in
https://opentrons-76.sentry.io/issues/6835148279/events/92c51920d13e42f68c4a59586ca32716/?project=4509363266387968,
we we crashing because `.tiprackDefURI` was **undefined** (rather than
just being an empty array).

I scoured the code for how `.tiprackDefURI` could be set to `undefined`,
and found only one potential place during `LOAD_FILE`:

https://github.com/Opentrons/opentrons/blob/ec8d0d6710d4dfee0b0a1335e584b89c270d499a/protocol-designer/src/step-forms/reducers/index.ts#L1247

In the protocol file, `designerApplication.pipetteTiprackAssignments` is
supposed to contain an entry for every pipette ID. I suspect what's
happening is that the pipette ID is simply missing in
`pipetteTiprackAssignments` (perhaps because the user deleted the
pipette, but we didn't completely eliminate all references to that
pipette in `savedStepForms` or `commands`).

If that's the case, the protocol file is malformed, but there's nothing
we can do about it. So this PR just defaults to an empty `tiprackDefURI`
if the pipette ID does not exist in `pipetteTiprackAssignments`.

## Test Plan and Hands on Testing

Smoke-tested on local PD.

## Risk assessment

Low.

jerader committed:
fix(app): blowout and disposal volume form fix (#19625)

closes RQA-4687

ddcc4 committed:
fix(protocol-designer): instrumentation to see why getMinXYDimension() is failing (#19626)

# Overview

We have a long-outstanding PD crash when we try to call
`getMinXYDimension()` in `SecondStepsMoveLiquidTools.tsx`, but I can't
figure out why it's failing:
https://opentrons-76.sentry.io/issues/6804827811/?project=4509363266387968

I'm not sure if it's because `formData['aspirate/dispense_labware']` is
blank, or if the labware is not in `labwareEntities`, or if the labware
is in `labwareEntities` but its `def` is missing for some reason.

This PR adds some instrumentation to try to catch the problem earlier,
and give us more information about why it's failing.

## Test Plan and Hands on Testing

Ran PD locally.

## Risk assessment

Low.

koji committed:
fix(app): fix air gap display issue in aspirate (#19601)

* fix(app): fix air gap display issue in aspirate

ddcc4 committed:
fix(protocol-designer): don't crash in volumeTooHigh() if tipRack is null (#19631)

# Overview

I have not been able to reproduce these PD crashes reported in Sentry:
- https://opentrons-76.sentry.io/issues/6867128274/?project=4509363266387968
- https://opentrons-76.sentry.io/issues/6867128191/?project=4509363266387968

But from the stack trace, they seem to be happening because we're
calling `volumeTooHigh()` with a null `tipRack` from the form, which in
turn calls `getPipetteCapacity()` and `getTiprackVolume()`, where we
crash.

Note we previously fixed the crash in `volumeTooHigh()` caused by
`pipette` being null (PR #19353), so we can fix the crash caused by
`tipRack` being null in the same way.

## Test Plan and Hands on Testing

Smoke tested PD running locally.

## Risk assessment

Low.

ncdiehl11 reviewed (approved):
Approving on behalf of transfer submerge, retract, and tip position quick transfer components

jbleon95 committed:
fix(api): correctly update and reset flow rates and restore pipette state after liquid class transfers (#19595)

Fixes a long-standing bug in the Python API code, where default flow rates would not correctly update in response to different capacity tips being picked up or volume mode being changed, and configure_for_volume wasn't resetting flow rates as the documentation claimed.

emilyburghardt committed:
PD 8.5.4 release notes (#19633)

<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure
you've read the "Opening Pull Requests" section of our Contributing
Guide:


https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams,
pictures, and videos along with text formatting make it possible to
create a rich and informative PR. For more information on GitHub
markdown, see:


https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out
the sections below to the best of your ability!
-->

# Overview

adding PD 8.5.4 release notes; includes a few bug fixes. 
<!--
Describe your PR at a high level. State acceptance criteria and how this
PR fits into other work. Link issues, PRs, and other relevant resources.
-->

## Test Plan and Hands on Testing

<!--
Describe your testing of the PR. Emphasize testing not reflected in the
code. Attach protocols, logs, screenshots and any other assets that
support your testing.
-->

## Changelog

<!--
List changes introduced by this PR considering future developers and the
end user. Give careful thought and clear documentation to breaking
changes.
-->

## Review requests

<!--
- What do you need from reviewers to feel confident this PR is ready to
merge?
- Ask questions.
-->

## Risk assessment

low. 
<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your
change may affect some other part of the system.
- For instance, changing return tip behavior may also change the
behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's
risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is
enough.
-->

ddcc4 committed:
Merge back 'chore_release-pd-8.5.4' into 'chore_release-8.7.0' (#19632)

Resolved conflicts:
- protocol-designer/src/step-forms/reducers/index.ts
